docs/authoring.md

```diff
@@ -259,3 +224,112 @@ There are some template parameters that have a special meaning:
+
+
+## React
+[React](https://reactjs.org) is the render engine which we strongly suggest you should use for any new templates. The only reason it is not the default render engine is to stay backward compatibility.
+* It enables the possiblity of [debugging](#debugging-react-template) your template (this is not possible with Nunjucks).
+* It provides better error stack traces once something is not working.
+* Provides better support for separating code into more manageable chunks/components.
+* The readability of the template are much better then Nunjucks syntax.
+* Better tool support for development.
+
+When writing react templates you decide whether to use CommonJS, ES5 or ES6 modules since everything is bundles together before the rendering process takes over. We use our own React renderer which can be found in the [Generator React SDK](https://github.com/asyncapi/generator-react-sdk). 
+There you can find information about how the renderer works or how we transpile your template files.
+
+Your React template always require 3 dependencies `react`, `source-map-support` and `@asyncapi/generator-react-sdk`. The `react` dependency is required to declare React components inside the template. `source-map-support` ensure that the transpiled files are mapped to the actual template files, this is only to enable correct error stack trace. `@asyncapi/generator-react-sdk` is required to access the `File` component which is required as a root component for a file to be rendered. Furthermore it provides some common components to make your development easier, like `Text` or `Indent`.
+
+Lets consider a basic react template as the one below called `MyTemplate.js`:
+```js
+import React from "react";
+import { File, Text } from "@asyncapi/generator-react-sdk";
+export default function({ asyncapi, params, originalAsyncAPI}) {
+  return (
+    <File name={'asyncapi.md'}>
+      <Text>Some text that should render as is</Text>
+    </File>
+  );
+}
+```
+The exported default function returns a `File` component as a root component which the [Generator](https://github.com/asyncapi/generator) uses to figure out what file should be generated. In our case we overwrite the default functionality of saving the file as `MyTemplate.js` but instead use the filename `asyncapi.md`. It is then specified that we should render `Some text that should render as is\n` within that file. Notice the `\n` character at the end, this is something that is automatically added after each children in the `File` and `Indent` component. 
+
+For further information about components, props etc. see the [Generator React SDK](https://github.com/asyncapi/generator-react-sdk)
+
+### Common assumptions
+
+1. All files within the template directory are rendered if their root component is of type `File` and the path is not part of the `nonRenderableFiles` configuration.
+1. New lines are automatically added for each child in `File` and `Indent` component.
```

Comment:
`New lines are automatically added for each child in `File` and `Indent` component.` is wrong.

New lines are only added in `Text` component. File is w wrapper and Indent adds only indentations (without new lines) :) https://github.com/asyncapi/generator-react-sdk/blob/1e897fc2d2d6b42fb2513f067d88bb5e9b8f2c88/src/utils/withIndendation.ts